### ISSUE TITLE
retrieve user password

### ISSUE THREAD

**Issue by ParmaU** [osTicket/osTicket-1.7#383]
I wanted to ask to integrate a feature where a user can recover your password
currently only an administrator can reset it

**Comment by greezybacon** [comment]
Passwords are never recoverable. Are you talking about a _forgot my password_ link?

**Comment by greezybacon** [comment]
@tratru do you need a different answer? Storing passwords in a reversible is a major security issue

**Comment by ParmaU** [comment]
my question is that you can retrieve or change your password without administrator.
with the usual method enter the e-mail send a link to the mail box for editing / recovery

**Comment by greezybacon** [comment]
I see, so you mean the user-level "forgot password ..." link. Se more 'reset' than 'recover' or 'retrieve'?

I agree that osTicket needs a password reset feature

**Comment by josh1856** [comment]
I also agree that osticket needs a password reset feature but there are other priorities.

**Comment by mckaygerhard** [comment]
ping...

i can made some workarounds but i dont understant how (or where) osticket made the string on the table "ost_user_account".passwd that are not an md5 string like staff table

how osticket generate this string?

**Comment by protich** [comment]
@mckaygerhard -- latest version of osTicket has password reset feature. Passwords are hashed, meaning it's not recoverable. 

**PS**: osTicket v1.7 series is coming to an end. It will no longer be supported come March 31, 2015.

**Comment by greezybacon** [comment]
@ParmaU can this issue be closed? User and Staff password reset has been implemented since 1.9.1 and 1.7.1 respectively.

**Comment by mckaygerhard** [comment]
ok, ok, so then i ask: there's a way to create a new password with sql commands (edited the question)! not for admin or agnets, i ask for users!
PD: this issue can be closed i think!

**Comment by roopesh007** [comment]
How to get forgot password feature enable in osticket, I am using osticket since 2 year and my users are not able to reset there password using forgot password option.

**Comment by igormop** [comment]
I create a link on page /include/client/login.inc.php like this:

<a href="pwreset.php"><?php echo __('Forget my pass'); ?></a>

Just below the Button Sign In, and it's working for  @ParmaU @roopesh007

**Comment by igormop** [comment]
I create a link on page /include/client/login.inc.php like this:

` <a href="pwreset.php"><?php echo __('S.O.S. with my pass :( '); ?></a>`

Just below the Button Sign In, and it's working for @ParmaU @roopesh007
